src/PrestaShopBundle/Controller/Admin/CategoryController.php

```diff
@@ -60,25 +65,43 @@ public function addSimpleCategoryFormAction(Request $request)
         if ($form->isValid()) {
             $data = $form->getData();
 
-            $_POST = [
-                'submitAddcategory' => 1,
-                'name_1' => $data['category']['name'],
-                'id_parent' => $data['category']['id_parent'],
-                'link_rewrite_1' => $tools->link_rewrite($data['category']['name']),
-                'active' => 1,
-                'checkBoxShopAsso_category' => $currentIdShop ? [$currentIdShop => $currentIdShop] : $shopList,
+            $localizedName = [
+                $defaultLanguageId => $data['category']['name'],
             ];
 
-            $adminCategoryController = $this->get('prestashop.adapter.admin.controller.category')->getInstance();
-            if ($category = $adminCategoryController->processAdd()) {
-                $response->setData(['category' => $category]);
-            }
+            $command = new AddCategoryCommand(
+                $localizedName,
+                [$defaultLanguageId => $tools->linkRewrite($data['category']['name'])],
+                true,
+                (int) $data['category']['id_parent']
+            );
+
+            $command->setAssociatedShopIds($currentIdShop ? [$currentIdShop => $currentIdShop] : $shopList);
+
+            try {
+                /** @var CategoryId $categoryId */
+                $categoryId = $commandBus->handle($command);
+
+                if ($categoryId->getValue()) {
+                    $response->setData(
+                        [
+                            'category' => [
+                                'id' => $categoryId->getValue(),
+                                'id_parent' => $data['category']['id_parent'],
+                                'name' => $localizedName,
+                            ],
+                        ]
+                    );
 
-            if ($request->query->has('id_product')) {
-                $productAdapter = $this->get('prestashop.adapter.data_provider.product');
-                $product = $productAdapter->getProduct($request->query->get('id_product'));
-                $product->addToCategories($category->id);
-                $product->save();
+                    if ($request->query->has('id_product')) {
+                        $productAdapter = $this->get('prestashop.adapter.data_provider.product');
+                        $product = $productAdapter->getProduct($request->query->get('id_product'));
+                        $product->addToCategories($categoryId->getValue());
+                        $product->save();
+                    }
+                }
+            } catch (CategoryException $e) {
+                // @todo error handling should be implemented.
```

Comment:
@jolelievre same as above, I suggest to not expand this PR with improvements as this is how the legacy code was behaving

Reply:
what's the point of migrating if we keep the legacy flaws? you can leave the `@todo` if you think we need to manage it better but I'm asking to add
`$this->flashErrors([$e->getMessage()]);` for now so that the user has a minimum of feedback

Reply:
But it's an ajax action, so `flashErrors` won't work as expected, we need to implement error handling in JS, which requires to change the legacy javascript

Reply:
My bad! I didn't see it was an ajax action. Let's do this later indeed, you are right